PhysicsTools/NanoAOD/python/globals_cff.py

```diff
@@ -31,9 +31,11 @@
     variables = cms.PSet(
         x1 = Var( "?hasPDF?pdf().x.first:-1", float, doc="x1 fraction of proton momentum carried by the first parton",precision=14 ),
         x2 = Var( "?hasPDF?pdf().x.second:-1", float, doc="x2 fraction of proton momentum carried by the second parton",precision=14 ),
-	#AR: not sure what "xPDF" is wrt to just "x"
-        #x1PDF = Var( "?hasPDF?pdf().xPDF.first:-1", float, doc="x1 fraction of proton momentum carried by the first parton" ),
-        #x2PDF = Var( "?hasPDF?pdf().xPDF.second:-1", float, doc="x2 fraction of proton momentum carried by the second parton" ),
+        xpdf1 = Var( "?hasPDF?pdf().xPDF.first:-1", float, doc="x1 fraction of proton momentum carried by the first parton", precision=14 ),
+        xpdf2 = Var( "?hasPDF?pdf().xPDF.second:-1", float, doc="x2 fraction of proton momentum carried by the second parton", precision=14 ),
+        id1 = Var( "?hasPDF?pdf().id.first:-1", int, doc="id fraction of proton momentum carried by the first parton", precision=6 ),
+        id2 = Var( "?hasPDF?pdf().id.second:-1", int, doc="id fraction of proton momentum carried by the first parton", precision=6 ),
+        scalePDF = Var( "?hasPDF?pdf().scalePDF:-1", float, doc="id fraction of proton momentum carried by the first parton", precision=14 ),
```

Comment:
can you fix the help strings?

Reply:
also can you check the difference between x and xpdf and understand which of the two is the one to keep? (i don't think we need both...)

Reply:
We need both, x is the momentum fraction, xpdf is "x * pdf(x)". However, it looks like they are often set to zero for our production.